### PR TITLE
Add 'sudo' to resolve Linux-related permission error

### PR DESCRIPTION
**Pull request by Ronit-j**
Signed-off-by: Ronit-j <pictronit007@gmail.com>

### CODE REVIEW

File: go/Makefile [hunk 1]
```diff
@@ -51,7 +51,7 @@ build-all:
 
 .PHONY: install
 install: build
-	cp $(BINARY) $(INSTALL_PATH)
+	sudo cp $(BINARY) $(INSTALL_PATH)
```
Comment (andreasjansson) [inline go/Makefile:54]:
```suggestion
# install without sudo if the install path exists and is writeable,
# or if it doesn't exist and its directory is writeable
	if [[ (-f "$(INSTALL_PATH)" && -w "$(INSTALL_PATH)") || (! -f "$(INSTALL_PATH)" && -w $$(dirname "$(INSTALL_PATH)")) ]]; then \
	    cp $(BINARY) $(INSTALL_PATH); \
	else \
	    sudo cp $(BINARY) $(INSTALL_PATH); \
	fi
```

**Review by andreasjansson — CHANGES_REQUESTED** [review]
Thanks for this @Ronit-j! Just added a suggestion to make the sudo install conditional on whether the file is writeable or not already.

File: go/Makefile (hunk 1 above)
Comment (Ronit-j) [reply go/Makefile:54]: Thanks @andreasjansson, I'll add it.

**Review by andreasjansson — APPROVED** [review]
Great, thank you! Merging.

**Pull request merged**